src/Abstractions/RetryPolicy.cs

```diff
@@ -39,7 +40,8 @@ public class RetryPolicy
         TimeSpan firstRetryInterval,
         double backoffCoefficient = 1.0,
         TimeSpan? maxRetryInterval = null,
-        TimeSpan? retryTimeout = null)
+        TimeSpan? retryTimeout = null,
+        Func<TaskFailedException, bool>? handle = null)
```

Comment:
Would it be simpler to use `TaskFailureDetails` instead of `TaskFailedException` for the parameter? It seems your test cases are only using the failure details anyways.

Reply:
I'm good with that change. Will do that tonight.

Any idea why the retry handler delegate would be getting called so many times? On a test case allowing ten retries, the handler delegate was called 108 times.

Reply:
Change to `TaskFailureDetails` is complete.